wordpress/wp-content/plugins/gc-lists/tests/Integration/Api/TestMessages.php

```diff
@@ -0,0 +1,104 @@
+	$this->assertIsArray($response->get_data());
+	$this->assertCount(5, $response->get_data());
+});
+
+test('Get sent messages', function() {
+	$template = $this->factory->message->create_and_get();
+
+	$this->factory->message->create_many(5, [
+		'original_message_id' => $template->id
+	]);
+
+	$request  = new WP_REST_Request( 'GET', '/gc-lists/messages/sent' );
+	$response = $this->server->dispatch( $request );
+
+	$this->assertEquals( 200, $response->get_status() );
+	$this->assertIsArray($response->get_data());
+	$this->assertCount(5, $response->get_data());
+});
+
+test('Get one messages', function() {
+	$message = $this->factory->message->create_and_get([
+		'name' => 'This is the message name'
+	]);
+
+	$request  = new WP_REST_Request( 'GET', "/gc-lists/messages/{$message->id}" );
+	$response = $this->server->dispatch( $request );
+
+	$this->assertEquals( 200, $response->get_status() );
+	$this->assertIsObject($response->get_data());
+	$this->assertObjectHasAttribute('name', $response->get_data());
+	$this->assertEquals('This is the message name', $response->get_data()->name);
+});
+
+test('Create a message', function() {
+	$request  = new WP_REST_Request( 'POST', '/gc-lists/messages' );
+	$request->set_query_params([
+		'name' => 'Name of the message',
+		'subject' => 'Subject of the message',
+		'body' => 'Body of the message',
+		'message_type' => 'email'
+	]);
+
+	$response = $this->server->dispatch( $request );
+
+	$this->assertEquals(200, $response->get_status());
+	$this->assertIsObject($response->get_data());
+	$this->assertObjectHasAttribute('name', $response->get_data());
+	$this->assertEquals('Name of the message', $response->get_data()->name);
+});
+
+test('Update a message', function() {
+	$message = $this->factory->message->create_and_get([
+		'name' => 'This is the message name'
+	]);
+
+	$this->assertEquals('This is the message name', $message->name);
+
+	$request  = new WP_REST_Request( 'PUT', "/gc-lists/messages/{$message->id}" );
+	$request->set_query_params([
+		'id' => $message->id,
+		'name' => 'Name of the message',
+		'subject' => 'Subject of the message',
+		'body' => 'Body of the message',
+	]);
+
+	$response = $this->server->dispatch( $request );
+
+	$this->assertEquals( 200, $response->get_status() );
+	$this->assertIsObject($response->get_data());
+	$this->assertObjectHasAttribute('name', $response->get_data());
+	$this->assertEquals('Name of the message', $response->get_data()->name);
+});
+
+test('Delete a message', function() {
+	$message_ids = $this->factory->message->create_many(5);
+
+	global $wpdb;
+	$count = $wpdb->get_var("SELECT COUNT(*) FROM {$wpdb->prefix}messages");
+	$this->assertEquals(5, $count);
+
+	$request  = new WP_REST_Request( 'DELETE', "/gc-lists/messages/{$message_ids[2]}" );
+	$response = $this->server->dispatch( $request );
+
+	$this->assertEquals( 200, $response->get_status() );
+	$count = $wpdb->get_var("SELECT COUNT(*) FROM {$wpdb->prefix}messages");
```

Comment:
Can we create a tablename var we can access for the tests vs `{$wpdb->prefix}messages}` in the tests